site/app/templates/notebook/Notebook.twig

```diff
@@ -262,6 +271,43 @@
 </div>
 
 <script>
+    //for displaying popup submit button
+    {% if is_timed and allowed_minutes > 25 %}
+        //interval is 20 percent of overall time allowed or 10 minutes, whichever is greater
+        let interval = Math.floor(((({{ allowed_minutes }}) / 5) > 10 ? ({{ allowed_minutes }}) / 5 : 10));
+        let loadTime = Date.now();
+        let showPopup = false;
+
+        function initializePopupTimer(){
+            sessionStorage.setItem("show_popup_submit", "yes");
+            count();
+        }
+
+        function count(){
+            const time_taken = Math.floor((curTime - loadTime)/1000);
+            const minutes_taken = Math.floor(time_taken / 60);
+            if (minutes_taken >= interval ) {
+                if (sessionStorage.getItem("show_popup_submit") === 'yes' && curTime < user_deadline && showPopup) {
+                    $(".notebook-submit-popup").show();
+                }
+            }
+            setTimeout(count,100);
```

Comment:
This will run every 0.1 seconds. I had this for timer for accuracy but for something like this it could probably run once a second or maybe even every few seconds.